molten-cache/src/main/java/com/hotels/molten/cache/metrics/CaffeineCacheStatsInstrumenter.java

```diff
@@ -109,16 +133,14 @@ public void recordEviction(int weight, RemovalCause cause) {
     @Nonnull
     @Override
     public CacheStats snapshot() {
-        return new CacheStats(Double.valueOf(hitCount.count()).longValue(), Double.valueOf(missCount.count()).longValue(), loadSuccess.count(), loadFailure.count(),
+        return CacheStats.of(Double.valueOf(hitCount.count()).longValue(), Double.valueOf(missCount.count()).longValue(), loadSuccess.count(), loadFailure.count(),
```

Comment:
Is this a compatibility breaking change with older caffeine? We should bump at least minor version then.

Reply:
Yes, caffeine needs to be updated if this is merged.
One can stay on the compatibility train, then need to update to 2.9.0 or later. No backward incompatible changes were done here by us of by caffeine.
One can switch to 3.0.0 or later, but then the incompatiblity needs to be issued.

Based on this, I agree to bump minor version. 👍🏻 

More details here:
https://github.com/ben-manes/caffeine/releases/tag/v3.0.0

Reply:
I mean when was `CacheStats.of` introduced? And is `CacheStats` constructor hidden now?
If yes, then it makes no sense to keep the above deprecated methods either since you won't be able to use this with caffeine 2.x anyway..

Reply:
It was introduced in 3.x train, but was merged back to 2.x train as well, so this won't prevent using this molten version with caffeine 2.x.